File: data/src/main/java/com/ykb/app/cryptotrader/data/repo/TradeBotRepo.java
package com.ykb.app.cryptotrader.data.repo;

import com.ykb.app.cryptotrader.data.model.TradeBot;
import com.ykb.app.cryptotrader.utils.enums.TradeBotStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface TradeBotRepo extends JpaRepository<TradeBot, Long> {

    @Query("SELECT t FROM TradeBot t WHERE t.status = TradeBotStatus.TERMINATED")
    List<TradeBot> findAllNonTerminated();

}

File: library/data/data-domain/src/main/kotlin/com/ykb/app/cryptotrader/data/domain/repo/TradeBotRepo.kt
package com.ykb.app.cryptotrader.data.domain.repo

import com.ykb.app.cryptotrader.data.domain.model.TradeBot
import com.ykb.app.cryptotrader.utils.enums.TradeBotStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface TradeBotRepo : JpaRepository<TradeBot, Int> {

    @Query("SELECT t FROM TradeBot t WHERE t.status = :status")
    fun findByStatus(status: TradeBotStatus): List<TradeBot>

}